src/Features/Core/Portable/Diagnostics/DiagnosticService.cs

```diff
@@ -150,12 +181,48 @@ private bool UpdateDataMap(IDiagnosticUpdateSource source, DiagnosticsUpdatedArg
             }
         }
 
+        private bool ClearDiagnosticsFromDataMapFor(IDiagnosticUpdateSource source, List<DiagnosticsUpdatedArgs> removed)
```

Comment:
>ClearDiagnosticsFromDataMapFor [](start = 21, length = 30)

Wouldn't `ClearDiagnosticsReportedBySource` be better name?

Reply:
sure.